docs/Testing.rst

```diff
@@ -33,6 +33,17 @@ The LLVM lit-based testsuite
 
 * Buildbots run all tests, on all supported platforms.
 
+Testsuite subsets
+-----------------
+
+The testsuite is split into three subsets:
+
+* Primary testsuite, located under ``swift/test``.
+
+* Validation testsuite, located under ``swift/validation-test``.
+
+* Long tests, which are marked with ``REQUIRES: long_test``.
```

Comment:
Perhaps add a blurb on how long a test can be expected to be in order to be categorized as a `long_test`?